packages/block-library/src/table/state.js

```diff
@@ -276,16 +276,25 @@ export function deleteColumn( state, {
  * @return {Object} New table state.
  */
 export function toggleSection( state, sectionName ) {
-	// Section exists, replace it with an empty row to remove it.
+	const sectionSaved = `${ sectionName }Saved`;
+
+	// Section exists, save it and replace it with an empty row to remove it.
 	if ( ! isEmptyTableSection( state[ sectionName ] ) ) {
-		return { [ sectionName ]: [] };
+		return { [ sectionName ]: [], [ sectionSaved ]: [ ...state[ sectionName ] ] };
```

Comment:
You can see here the values returned by `toggleSection` get saved as attributes:
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/table/edit.js#L239

This kind of transient state isn't really the intended use case for the block attributes. I think ideally `headSaved` and `footSaved` would instead be stored in component state.

Reply:
Yep I noticed that when Travis ran the tests, the attributes object seems to be excpected to be always having the same structure without any extra props.